src/domain/GetNotificationDocumentMetadataRepository.ts

```diff
@@ -0,0 +1,26 @@
+import { ApiKey } from '../generated/definitions/ApiKey';
+import { Iun } from '../generated/definitions/Iun';
+import { NotificationAttachmentDownloadMetadataResponse } from '../generated/definitions/NotificationAttachmentDownloadMetadataResponse';
+import { Repository } from './Repository';
+import { Response, UnauthorizedMessageBody } from './types';
+import { Notification } from './Notification';
+
+export type GetNotificationDocumentMetadataRecord = {
+  type: 'GetNotificationDocumentMetadataRecord';
+  input: { apiKey: ApiKey; iun: Iun; docIdx: number };
+  output:
+    | Response<200, NotificationAttachmentDownloadMetadataResponse>
+    | Response<403, UnauthorizedMessageBody>
```

Comment:
I leave this comment here, just as a reminder and we can discuss about it in an appropriate meeting.

It looks like that now, if you make a request without the `x-api-key` header, the server returns a 401; if you make a call with the header, but with an invalid value, the server returns 403.

Here an example of a request that returns a 401:
```
curl --location --request GET 'https://api.dev.pn.pagopa.it/delivery/notifications/sent/6002032.770410135/attachments/documents/1' \
--header 'Accept: application/json'
```

Here an example of a request that returns a 403:
```
curl --location --request GET 'https://api.dev.pn.pagopa.it/delivery/notifications/sent/6002032.770410135/attachments/documents/1' \
--header 'Accept: application/json' \
--header 'x-api-key: asd'
```

Reply:
We should open a ticker about this mismatch

Reply:
This activity is tracked with [this](https://pagopa.atlassian.net/browse/VL-115) issue.